app/forms/openid_connect_authorize_form.rb

```diff
@@ -299,7 +312,8 @@ def scopes
   def validate_privileges
     if (identity_proofing_requested? && !identity_proofing_service_provider?) ||
        (ialmax_requested? && !ialmax_allowed_for_sp?) ||
-       (biometric_ial_requested? && !service_provider.biometric_ial_allowed?)
+       (biometric_ial_requested? && !service_provider.biometric_ial_allowed?) ||
+       (semantic_authn_contexts_requested? && !service_provider&.semantic_authn_contexts_allowed?)
```

Comment:
[thought/nonblocking] hm i know you didn't introduce the nil checking in this method, but i think it can be removed -- this validation runs after `ialmax_allowed_for_sp` which ensures that a service_provider exists. we can clean that up in a follow-up pr

Reply:
done!